torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -334,20 +338,25 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
 
   // Try moving persistent buffer factors into threads until we have too many
   // threads.
+  constexpr int batches_per_block_inner_reduction_max = 10;
   while (
-      // If using less than a quarter of available threads
-      bdimx * bdimy * bdimz * 2 <=
-          ceilDiv(device_max_threads_per_multiprocessor, (int64_t)4) &&
+      // If using less than one warp per scheduler
+      bdimx * bdimy * bdimz * 2 <= max_threads_in_block &&
```

Comment:
Is this comment still accurate?

Reply:
If `bdimx * bdimy * bdimz * 2 <= max_threads_in_block`, then shouldn't we decrease `bdimx`, `bdimy`, or `bdimz` no matter what the persistent buffer size is. We are setting `max_threads_in_block` somewhere beginning of this function, so if we are violating the maximum set before, it seems we should reduce the number of threads here irrespective of the buffer size, or we should change `max_threads_in_block`

Reply:
corrected.